llvm/include/llvm/MC/DXContainerRootSignature.h

```diff
@@ -7,19 +7,42 @@
 //===----------------------------------------------------------------------===//
 
 #include "llvm/BinaryFormat/DXContainer.h"
-#include <cstdint>
-#include <limits>
+#include "llvm/Support/BinaryStreamWriter.h"
+#include "llvm/Support/raw_ostream.h"
+#include <map>
+#include <string>
```

Comment:
We probably don't want to use `std::map` and `std::string` here - see the section on [picking-the-right-data-structure-for-a-task](https://llvm.org/docs/ProgrammersManual.html#picking-the-right-data-structure-for-a-task) from the LLVM programmer's manual for some good guidelines.